tests/FilterTest.php

```diff
@@ -335,37 +335,66 @@ public function testAppendThrowsShouldTriggerEndButIgnoreExceptionDuringEnd()
         $this->assertContainsString('test', $errors[0]);
     }
 
-    public function testAppendInvalidStreamIsRuntimeError()
+    public function testAppendInvalidStreamIsRuntimeErrorWithoutCallingCustomErrorHandler()
     {
         if (PHP_VERSION >= 8) $this->markTestSkipped('Not supported on PHP 8+ (PHP 8 throws TypeError automatically)');
 
         $this->setExpectedException('RuntimeException');
         if (defined('HHVM_VERSION')) $this->markTestSkipped('Not supported on HHVM (does not reject invalid stream)');
+
+        $error = null;
+        set_error_handler(function ($_, $errstr) use (&$error) {
+            $error = $errstr;
+        });
+
         StreamFilter\append(false, function () { });
+
+        restore_error_handler();
+        $this->assertNull($error);
     }
 
-    public function testPrependInvalidStreamIsRuntimeError()
+    public function testPrependInvalidStreamIsRuntimeErrorWithoutCallingCustomErrorHandler()
     {
         if (PHP_VERSION >= 8) $this->markTestSkipped('Not supported on PHP 8+ (PHP 8 throws TypeError automatically)');
 
         $this->setExpectedException('RuntimeException');
         if (defined('HHVM_VERSION')) $this->markTestSkipped('Not supported on HHVM (does not reject invalid stream)');
+
+        $error = null;
+        set_error_handler(function ($_, $errstr) use (&$error) {
+            $error = $errstr;
+        });
+
         StreamFilter\prepend(false, function () { });
+
+        restore_error_handler();
+        $this->assertNull($error);
     }
 
-    public function testRemoveInvalidFilterIsRuntimeError()
+    public function testRemoveInvalidFilterIsRuntimeErrorWithoutCallingCustomErrorHandler()
     {
         if (PHP_VERSION >= 8) $this->markTestSkipped('Not supported on PHP 8+ (PHP 8 throws TypeError automatically)');
 
         $this->setExpectedException('RuntimeException', 'Unable to remove filter: stream_filter_remove() expects parameter 1 to be resource, ');
         if (defined('HHVM_VERSION')) $this->markTestSkipped('Not supported on HHVM (does not reject invalid filters)');
+
+        $error = null;
+        set_error_handler(function ($_, $errstr) use (&$error) {
+            $error = $errstr;
+        });
+
         StreamFilter\remove(false);
+
+        $error = null;
+        set_error_handler(function ($_, $errstr) use (&$error) {
+            $error = $errstr;
+        });
     }
 
     /**
      * @depends testAppendEndEventWillBeCalledOnRemove
      */
-    public function testRemoveThrowsWhenAppendThrowsOnEvent()
+    public function testRemoveThrowsWhenAppendThrowsOnEventWithoutCallingCustomErrorHandler()
```

Comment:
Copy pasta?

Reply:
Don't know what I was thinking here but this is clearly wrong.